### ISSUE TITLE
Callback order in fit_loop

### ISSUE THREAD

**Issue by ELind77** [keras-team/keras#10691]
Hi all,

tl;dr: Is `History` last in the list of callbacks for a reason?

I've been implementing a custom callback to produce a training report at the end of each epoch during model training.  As part of my report I plot the training and validation loss so far.  In my implementation, I thought it would make things easier to use the existing history callback rather than accumulate my own list of losses, as `History` is always guaranteed to be present.  However, because the history callback is always last in the list (seen [here](https://github.com/keras-team/keras/blob/master/keras/engine/training_arrays.py#L104)) I have to go through a few extra steps to get the behavior I want.  While this isn't a huge problem it led me to wonder if there was a specific reason for making `History` the last callback in the list.  I didn't see any comments in the source so I thought I'd ask here.

-- Eric
